sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala

```diff
@@ -171,6 +172,13 @@ sealed abstract class CoalesceSizeGoal extends CoalesceGoal {
  */
 trait RequireSingleBatchLike
 
+/**
+ * Trait used for pattern matching for goals that could be split, as they
+ * only specify that batches won't be too much bigger than a maximum target
+ * size in bytes.
+ */
+trait SplittableGoal
```

Comment:
Do we have a follow on issue for BatchedByKey yet?

Reply:
yes we split this off early on https://github.com/NVIDIA/spark-rapids/issues/7778